app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -257,6 +264,37 @@ class NavigationOverlayFragment : Fragment() {
         activity!!.moveTaskToBack(true)
     }
 
+    private fun observeAccountState(): Disposable {
+        val fxaRepo = serviceLocator.fxaRepo
+
+        return fxaRepo.accountState.subscribe { accountState ->
+            when (accountState) {
+                is AccountState.AuthenticatedWithProfile -> {
+                    val fxaProfile = accountState.profile
+                    if (fxaProfile.avatar != null) {
+                        PicassoWrapper.client
```

Comment:
wrt my "explicitly handling all states" comments, one way to make sure we're performing all assignments in all states is to avoid duplicating the final UI update code:
```kotlin
val contentDescription: String
val avatarPath: String
when (accountState) {
    is AccountState.AuthenticatedWithProfile -> {
        contentDescription = ...
        avatarPath = ...
    }
    ...
}

PicassoWrapper.client.load(avatarPath).into(fxaButton)
fxaButton.contentDescription = contenDescription
```

Then, when a new state is added, the compiler checks that they made the assignment in their new state and the UI update code will happen automatically.

In practice, it doesn't always work out easily because the behavior may be different (e.g. setting image from resource vs. from URL) but you can often use lambdas and other types, like our `ImageSetStrategy`, to make it work.

I suggest you use this pattern here – it's ideal – but you don't have to.